crates/script/src/receipts.rs

```diff
@@ -37,7 +37,27 @@ pub async fn check_tx_status(
                 .get_receipt()
                 .await
             {
-                Ok(receipt) => Ok(receipt.into()),
+                Ok(receipt) => {
+                    // Check if the receipt has valid block information
+                    if receipt.block_number.is_none() || receipt.block_hash.is_none() || receipt.transaction_index.is_none() {
```

Comment:
receipt usually should only return mined transactions, but I can see that some networks might behave that way.

since these are mostly faster L2s, I wonder if the better approach here would be, just retry a few times here?